From b78ea7f9f91582cca8da9e13f6060e4e74661dff Mon Sep 17 00:00:00 2001
From: Dicky Chiang <dickychiang73@gmail.com>
Date: Sun, 23 Oct 2011 03:11:26 +0800
Subject: [PATCH] fix wifi enable disable issue

Cause is ehci USB on use it when kernel init. Rerfer the datasheet of DM37x :

-------------------------------------------------------------------------------------
Mode0    Mode1    Mode2      Mode3       Mode4    Mode5          Mode6        Mode7
-----  --------- ------ ---------------------- ---------- ----------------- -------
etk_d2 mcspi3_cs         hsusb1_data2  gpio_16  mm1_txdat   hsusb1_tll_data2  hw_dbg4
-------------------------------------------------------------------------------------

usb-ehci will be config to Mode3 and voltage is always pulled.so need to diable usb-echi[0]
---
 arch/arm/mach-omap2/board-omap3beagle.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/board-omap3beagle.c b/arch/arm/mach-omap2/board-omap3beagle.c
index aa03042..44bfe84 100644
--- a/arch/arm/mach-omap2/board-omap3beagle.c
+++ b/arch/arm/mach-omap2/board-omap3beagle.c
@@ -868,7 +868,7 @@ static void __init omap3beagle_flash_init(void)
 
 static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 
-	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
+	.port_mode[0] = EHCI_HCD_OMAP_MODE_UNKNOWN,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[2] = EHCI_HCD_OMAP_MODE_UNKNOWN,
 
-- 
1.7.4.1

